### PR TITLE
Tests for astronomy load done : Caleb

### LINKED ISSUE #202: Create test file for the weekly astronomy load script
As a developer, I want to test the `astronomy_load.py` file to ensure that the data is being loaded correctly and robustly, and that the functions deal with appropriate edge cases and erroneous input effectively so that the user experience is as intended.

### LINKED ISSUE #203: Write tests for the weekly astronomy load script
As a developer, I want to test the `astronomy_load.py` file to ensure that the data is being loaded correctly and robustly, and that the functions deal with appropriate edge cases and erroneous input effectively so that the user experience is as intended.

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #202 
This closes #203 

Made the file and added tests for astronomy load, added a few fixtures to conftest also.

### CODE REVIEW

File: astronomy-pipeline/conftest.py [hunk 1]
```diff
@@ -107,3 +107,36 @@ def sample_raw_positions():
...
+@pytest.fixture
+def upload_body_data_query():
+    '''Body data sql query for testing.'''
+    q_str = """
+                INSERT INTO body_assignment
+                (at, distance_km, azimuth, altitude, region_id, body_id, constellation_id)
+                VALUES
+                %s"""
+
+    return q_str
```
Comment (AlexOBarnes) [inline astronomy-pipeline/conftest.py:121]: could be renamed to be more fitting

File: astronomy-pipeline/test_astronomy_load.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    @patch('astronomy_load.execute_values')
+    @patch('astronomy_load.get_db_connection')
+    def test_upload_body_position_data_success_functions(self, mock_get_db_connection,
+                                                         mock_execute_values):
+        '''Tests that the correct functions and methods are called.'''
+        mock_conn = MagicMock()
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+        upload_body_position_data([[]])
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_load.py:21]: Good use of mocking

File: astronomy-pipeline/test_astronomy_load.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        upload_body_position_data([[]])
+
+        mock_execute_values.assert_called_once_with(
+            mock_cursor, upload_body_data_query, [[]])
+
+
+class TestUploadMoonPhaseData():
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_load.py:43]: Good for dividing up the tests

File: astronomy-pipeline/test_astronomy_load.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        mock_conn.cursor.return_value = mock_cursor
+
+        upload_moon_phase_data([[]])
+
+        mock_execute_values.assert_called_once_with(
+            mock_cursor, upload_moon_data_query, [[None, None, 'moon_phase']])
+
+
+class TestUploadAstronomyData():
+    '''Tests for the upload astronomy data function.'''
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_load.py:78]: Love the comments

File: astronomy-pipeline/conftest.py [hunk 5]
```diff
@@ -107,3 +107,36 @@ def sample_raw_positions():
             }
         }
     }
+
+
+@pytest.fixture
```
Comment (AlexOBarnes) [inline astronomy-pipeline/conftest.py:112]: Good use of fixtures

**Review by AlexOBarnes — APPROVED** [review]
Looks good to me

File: astronomy-pipeline/conftest.py (hunk 1 above)
Comment (olliejth) [reply astronomy-pipeline/conftest.py:121]: Maybe "body_position_rds_data_query"?

File: astronomy-pipeline/conftest.py [hunk 6]
```diff
@@ -107,3 +107,36 @@ def sample_raw_positions():
...
+                (at, distance_km, azimuth, altitude, region_id, body_id, constellation_id)
+                VALUES
+                %s"""
+
+    return q_str
+
+
+@pytest.fixture
+def upload_moon_data_query():
+    '''Moon phase data sql query for testing.'''
```
Comment (olliejth) [inline astronomy-pipeline/conftest.py:126]: Specify "moon_phase"

File: astronomy-pipeline/test_astronomy_load.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    def test_upload_moon_phase_data_success_calls(self, mock_get_db_connection,
+                                                  mock_execute_values, upload_moon_data_query):
+        '''Tests that the execute values method is called with the correct inputs.'''
+        mock_conn = MagicMock()
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        upload_moon_phase_data([[]])
+
```
Comment (olliejth) [inline astronomy-pipeline/test_astronomy_load.py:72]: Good idea using an empty list to keep the tests as lightweight as possible.

File: astronomy-pipeline/test_astronomy_load.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    @patch('astronomy_load.upload_moon_phase_data')
+    @patch('astronomy_load.upload_body_position_data')
+    def test_upload_astronomy_data_success_calls_body(self, mock_upload_body_position_data,
+                                                      mock_upload_moon_phase_data,
+                                                      astronomy_data_dict):
+        '''Tests that upload body position data is called with the correct values.'''
+        upload_astronomy_data(astronomy_data_dict)
+
+        mock_upload_body_position_data.assert_called_once_with(['positions'])
```
Comment (olliejth) [inline astronomy-pipeline/test_astronomy_load.py:109]: Comprehensive test suite.

**Review by olliejth — APPROVED** [review]
Good robust test suite, could be developed in the future to include more edge cases?
